samples/tab-nested-auth/nodejs/src/components/Msal.js

```diff
@@ -0,0 +1,143 @@
+
+const Msal = () => {
+
+    const [meData, setMeData] = useState(null); // State to store user data
+    const [isLoggedIn, setIsLoggedIn] = useState(false); // State to track login status
+
+    let pca = undefined; // Public client application instance
+
+    // MSAL configuration
+    const msalConfig = {
+        auth: {
+            clientId: "{{clientId}}",
+            authority: "https://login.microsoftonline.com/common"
+        }
+    };
+
+    // Function to initialize the public client application
+    function initializePublicClient() {
+        console.log("Starting initializePublicClient");
+        return createNestablePublicClientApplication(msalConfig).then(
+            (result) => {
+                console.log("Client app created");
+                pca = result;
+                return pca;
+            }
+        );
+    }
+
+    // Function to get the active account
+    const getActiveAccount = async () => {
+        console.log("Starting getActiveAccount");
+        let activeAccount = null;
+        try {
+            console.log("Getting active account");
+            activeAccount = pca.getActiveAccount();
+        } catch (error) {
+            console.log(error);
+        }
+        if (!activeAccount) {
+            console.log("No active account, trying login popup");
+            try {
+                const context = await app.getContext();
+                const accountFilter = {
+                    tenantId: context.user?.tenant?.id,
+                    homeAccountId: context.user?.id,
+                    loginHint: (await app.getContext()).user?.loginHint
+                };
+                const accountWithFilter = pca.getAccount(accountFilter);
+                if (accountWithFilter) {
+                    activeAccount = accountWithFilter;
+                    pca.setActiveAccount(activeAccount);
+                }
+            } catch (error) {
+                console.log(error);
+            }
+        }
+        return activeAccount;
+    }
+
+    // Function to get the token for the user
+    const getToken = async () => {
+        let activeAccount = await getActiveAccount();
+        const tokenRequest = {
+            scopes: ["User.Read"],
+            account: activeAccount || undefined,
+        };
+        return pca.acquireTokenSilent(tokenRequest)
+            .then((result) => {
+                console.log(result);
+                return result.accessToken;
+            })
+            .catch((error) => {
+                console.log(error);
+                // Try to get token via popup if silent acquisition fails
+                return pca.acquireTokenPopup(tokenRequest)
+                    .then(async (result) => {
+                        console.log(result);
+                        return result.accessToken;
+                    })
+                    .catch((error) => {
+                        console.log(error);
+                        return JSON.stringify(error);
```

Comment:
I wouldn't return the error here, it's just going to cause an error later in the flow when we try to use the stringified error as a token. Would it be appropriate to just rethrow the error or return `undefined`?

Reply:
Done